core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -132,16 +134,20 @@ class GroupCoordinator(val brokerId: Int,
           responseCallback(joinError(memberId, Errors.UNKNOWN_MEMBER_ID.code))
         } else {
           group = coordinatorMetadata.addGroup(groupId, protocolType)
-          doJoinGroup(group, memberId, sessionTimeoutMs, protocolType, protocols, responseCallback)
+          doJoinGroup(group, memberId, clientId, clientHost, sessionTimeoutMs, protocolType,
+            protocols, responseCallback)
         }
       } else {
-        doJoinGroup(group, memberId, sessionTimeoutMs, protocolType, protocols, responseCallback)
+        doJoinGroup(group, memberId, clientId, clientHost, sessionTimeoutMs, protocolType,
+          protocols, responseCallback)
       }
     }
   }
 
   private def doJoinGroup(group: GroupMetadata,
                           memberId: String,
+                          clientId: String,
+                          clientHost: String,
```

Comment:
@hachikuji In KAFKA-2017 / KAFKA-2018 I made the clientId as part of the memberId, but maybe we can include both id / host as the memberId? With this do we still need the UUID even if we reset generation ids?